pom.xml

```diff
@@ -140,6 +140,12 @@
         <scope>test</scope>
         <classifier>tests</classifier>
       </dependency>
+      <dependency>
+        <groupId>org.jenkinsci.plugins</groupId>
+        <artifactId>pipeline-model-definition</artifactId>
+        <version>1.2.7</version>
```

Comment:
This issue happens only when using declarative pipelines.
Using the greatest version that doesn't cause `RequireUpperBoundDeps` for `workflow-multibranch`.
We can use a newer version by upgrading also the version of `workflow-multibranch`, but this version is enough to reproduce the issue.